## Review: Bounce-Back Cycle 1

**Reason:** CI failed (unrelated, will rebase)

**Reviewer A (Correctness):** APPROVE — refactor verified as no-op on the apply path; synthetic entry's `hash` is never read on case-4 (Apply) branch; clippy fix is semantically equivalent.

**Reviewer B (Risk):** APPROVE — synthetic `virtual_entry` is local-scope only, no API/format/log/metric surface touched, parity & determinism unaffected.

**CI:** red (unrelated)

The `Test` job in workflow `CI` (run [25946394786](https://github.com/stellar-experimental/henyey/actions/runs/25946394786/job/76275283319)) was cancelled at 00:02:55Z after ~30 min wall-clock because an unrelated test hung for >12 minutes:

```
2026-05-15T23:50:52Z test tx_set_tracker::tests::test_request_at_cap_still_increments_existing has been running for over 60 seconds
2026-05-16T00:02:55Z ##[error]The operation was canceled.
```

`tx_set_tracker` lives outside the diff (PR only touches `crates/history/src/catchup/replay.rs` and `crates/history/src/verify.rs`), so this failure is not diff-attributable — it's a hanging test in `henyey-herder` (or wherever `tx_set_tracker` lives) that timed out the job.

Routing back to `ready-for-doing` for `/do` Mode B to rebase on `origin/main` and re-run CI. If the same unrelated failure reappears after rebase, the next `/review-pr` will mark `blocked` and the hang must be investigated as a separate issue.

--- a/crates/history/src/catchup/replay.rs
+++ b/crates/history/src/catchup/replay.rs
@@ -185,14 +185,12 @@ impl CatchupManager {
         }
         if let Some(first) = apply_data.first() {
             let header = first.header().clone();
-            let entry_hash = henyey_ledger::compute_header_hash(&header).map_err(|e| {
-                HistoryError::CatchupFailed(format!(
-                    "Failed to compute header hash for ledger {}: {}",
-                    header.ledger_seq, e
-                ))
-            })?;
+            // `knit_to_lcl_decision` only reads `header.previous_ledger_hash` on
+            // the case-4 (Apply) branch exercised here, so the entry hash is
+            // unused. Use a zero placeholder rather than recomputing the header
+            // hash for a synthetic entry that we discard immediately.
             let virtual_entry = LedgerHeaderHistoryEntry {
-                hash: entry_hash.into(),
+                hash: stellar_xdr::curr::Hash([0; 32]),
                 header,
                 ext: Default::default(),
             };

--- a/crates/history/src/verify.rs
+++ b/crates/history/src/verify.rs
@@ -2204,7 +2204,7 @@ mod tests {
         // doesn't match hash(lower[126]).
         upper[0].previous_ledger_hash = stellar_xdr::curr::Hash([0xDD; 32]);
 
-        let mut headers: Vec<_> = lower.into_iter().chain(upper.into_iter()).collect();
+        let mut headers: Vec<_> = lower.into_iter().chain(upper).collect();
         // Re-compute internal chain for upper group: header 129's prev must
         // match hash of (tampered) header 128, etc. Use make_chain's approach.
         for i in 128..headers.len() {

